lab_3/main.py

```diff
@@ -17,39 +18,63 @@ def tokenize_by_sentence(text: str) -> tuple:
          (('_', 'h', 'e', '_'), ('_', 'i', 's', '_'), ('_', 'h', 'a', 'p', 'p', 'y', '_'))
          )
     """
-    pass
+    if not isinstance(text, str) or len(text) == 0:
+        return ()
+    tokens = re.split(r'[.?!] ', text)
+    list_of_tokens = []
+    for token in tokens:
+        token_list = re.sub('[^a-z \n]', '', token.lower()).split()
+        if not token_list:
+            return ()
+        list_of_tokens.append(tuple(tuple(['_'] + list(word) + ['_']) for word in token_list))
+    return tuple(list_of_tokens)
+
 
 
 # 4
 class LetterStorage:
 
     def __init__(self):
-        pass
+        self.storage = {}
 
     def _put_letter(self, letter: str) -> int:
         """
         Puts a letter into storage, assigns a unique id
         :param letter: a letter
         :return: 0 if succeeds, 1 if not
         """
-        pass
+        if not isinstance(letter, str) or len(letter) == 0:
+            return 1
+        value = 1
```

Comment:
(this variable is local)
if I want to add several words using _put_letter, will they have different or similar ids?